webapp/src/components/backstage/playbook_runs/playbook_run_backstage/metrics/metrics_data.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {forwardRef, useImperativeHandle, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import {VerticalSpacer} from 'src/components/backstage/playbook_runs/shared';
+
+import {RunMetricData} from 'src/types/playbook_run';
+import {Metric, MetricType} from 'src/types/playbook';
+import {ClockOutline, DollarSign, PoundSign} from '../../../playbook_edit/styles';
+import {isMetricValueValid, stringToMetric, metricToString} from '../../../playbook_edit/metrics/shared';
+
+import {useClickOutsideRef} from 'src/hooks';
+
+import MetricInput from './metric_input';
+
+interface MetricsProps {
+    metricsData: RunMetricData[];
+    metricsConfigs: Metric[];
+    isPublished: boolean;
+    onEdit: (metricsData: RunMetricData[]) => void;
+    flushChanges: () => void;
+}
+
+const MetricsData = forwardRef(({metricsData, metricsConfigs, isPublished, onEdit, flushChanges}: MetricsProps, ref) => {
+    const {formatMessage} = useIntl();
+
+    const initialValues = new Array(metricsConfigs.length);
+    metricsConfigs.forEach((mc, index) => {
+        const md = metricsData.find((metric) => metric.metric_config_id === mc.id);
+        initialValues[index] = md ? metricToString(md.value, mc.type) : '';
+    });
+    const [inputsValues, setInputsValues] = useState(initialValues);
+    const [inputsErrors, setInputsErrors] = useState(new Array(metricsConfigs.length).fill(''));
+
+    // Handles click outside of metrics inputs to save changes
+    const inputRef = useRef(null);
+    useClickOutsideRef(inputRef, () => {
+        flushChanges();
+    });
+
+    //  validateInputs function is called from retrospective component on publish button click, to validate metrics inputs
+    useImperativeHandle(
+        ref,
+        () => ({
+            validateInputs() {
+                const errors = verifyInputs(inputsValues, true);
+                setInputsErrors(errors);
+
+                return !hasErrors(errors);
+            },
+        }),
+    );
+
+    const errorCurrencyInteger = formatMessage({defaultMessage: 'Please enter a number, or leave the target blank.'});
```

Comment:
```suggestion
    const errorCurrencyInteger = formatMessage({defaultMessage: 'Please enter a number.'});
```
Because they aren't allowed to enter a blank metric. :)

Reply:
Nice catch : )